pkg/js/helpers.js

```diff
@@ -156,6 +170,18 @@ function ALIAS(name, target) {
     }
 }
 
+// CAA(name,tag,value, recordModifiers...)
+function CAA(name, tag, value){
+    checkArgs([_.isString, _.isString, _.isString], arguments, "CAA expects (name, tag, value)")
```

Comment:
I think I was unclear. I was referring to the error message.

    checkArgs([_.isString, _.isString, _.isString], arguments, "CAA expects (label, tag, target) plus optional flag as a meta argument")

That makes me think....

Usually the order of a record's data is "label", priorities and stuff (if it exists), "target", { "meta": "value"}. This is the first record there is data after the target that isn't metadata.  

On the one hand, it seems to be setting a bad precedent to have actual RFC-defined elements of the RR after the target.  On the other hand, it seems like the "flag" is something that will be rarely used, and should stand out visually (like by adding CAA_CRITICAL at the end).

So it comes down to the decision between these two syntaxes:

    CAA("@",  flag, "tag", "target", {meta:value))

or

    CAA("@",  "tag", "target", flag, {meta:value})

I'm not sure which is better.

One that thing would help us decide:

  * How frequently do you think CAA_CRITICAL will be used?
  * What syntax do you think will be the most natural to novices?
  * What will make the most sense to someone trying to translate between ISP/Cloud/Cert provider documentation and the DNSControl DSL?  For example, if someone reads this page https://support.google.com/a/answer/174125?hl=en it is pretty obvious what to do with MX().  Eventually providers will make similar documents explaining what to do with CAA record.  If you were to write a hypothetical document, what would it look like?  Would it easily translate into CAA() format?

CC: @captncraig